packages/terra-form/src/components/Input.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { PropTypes } from 'react';
+import classNames from 'classnames';
+import 'terra-base/lib/baseStyles';
+
+import './Input.scss';
+
+const propTypes = {
+  /**
+   * The defaultValue of the input field. Use this to create an uncontrolled input
+   */
+  defaultValue: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number,
+  ]),
+  /**
+   * Name of the input
+   */
+  name: PropTypes.string,
+  /**
+   * Functional to be called when the input is changed
+   */
+  onChange: PropTypes.func,
+  /**
+   * Whether the input is required or not
+   */
+  required: PropTypes.bool,
+  /**
+   * The value of the input field. Use this to create a controlled input
+   */
+  value: PropTypes.oneOfType([
+    PropTypes.string,
+    PropTypes.number,
+  ]),
+};
+
+const defaultProps = {
+  defaultValue: undefined,
+  name: null,
+  onChange: () => {},
```

Comment:
We have the need to support both controlled and uncontrolled inputs, so we can't always expect the developer to provide an onChange function.

One option is to just put it on the inputAttrs attribute for forms, and delegate it that way to the input element.